test/defaults.test.ts

```diff
@@ -147,4 +147,61 @@ describe("graphql.defaults()", () => {
       }
     }`);
   });
+
+  it("set defaults on .endpoint", () => {
+    const mockData = {
+      repository: {
+        issues: {
+          edges: [
+            {
+              node: {
+                title: "Foo",
+              },
+            },
+            {
+              node: {
+                title: "Bar",
+              },
+            },
+            {
+              node: {
+                title: "Baz",
+              },
+            },
+          ],
+        },
+      },
+    };
+
+    const authenticatedGraphql = graphql.defaults({
+      headers: {
+        authorization: `token secret123`,
+      },
+      request: {
+        fetch: fetchMock.sandbox().post(
+          "https://github.acme-inc.com/api/graphql",
+          { data: mockData },
+          {
+            headers: {
+              authorization: "token secret123",
+            },
+          }
+        ),
+      },
+    });
+
+    const { request: _request, ...requestOptions } =
+      // @ts-expect-error - TODO: expets to set { url } but it really shouldn't
```

Comment:
given that this never came up in years of the existence of this package, I don't think it's worth digging into it. I'll fix it in octokit-next.js though, the types work differently there though, so the same fix might not be applicable.

Reply:
```suggestion
      // @ts-expect-error - TODO: expects to set { url } but it really shouldn't
```